runtime/compiler/optimizer/MethodHandleTransformer.cpp

```diff
@@ -440,6 +440,11 @@ TR_MethodHandleTransformer::visitStoreToLocalVariable(TR::TreeTop* tt, TR::Node*
    {
    TR::Node *rhs = node->getFirstChild();
    TR::Symbol *local = node->getSymbolReference()->getSymbol();
+
+   // if we are storing the result of a call that is transformed into PassThrough
+   if (rhs->getOpCodeValue() == TR::PassThrough)
+      rhs = node->getFirstChild()->getFirstChild();
```

Comment:
To properly handle `PassThrough`, I think we need to loop until we find a child that isn't itself `PassThrough`, since `PassThrough` nodes could potentially be nested. Though I don't think the main change in this PR is likely to create such a situation. There would need to be something else also creating `PassThrough` before this point.